src/azure-cli/azure/cli/command_modules/resource/_bicep.py

```diff
@@ -0,0 +1,148 @@
+from knack.util import CLIError
+
+# See: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+_semver_pattern = r"(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?"  # pylint: disable=line-too-long
+_logger = get_logger(__name__)
+
+
+def run_bicep_command(args, auto_install=True, check_upgrade=True):
+    installation_path = _get_bicep_installation_path(platform.system())
+    installed = os.path.isfile(installation_path)
+
+    if not installed:
+        if auto_install:
+            ensure_bicep_installation()
+        else:
+            raise CLIError('Bicep CLI not found. Install it now by running "az bicep install".')
+    elif check_upgrade:
+        with suppress(CLIError):
+            # Checking upgrade should not raise CLIError.
+            # Users may continue using the current installed version.
+            installed_version = _get_bicep_installed_version(installation_path)
+            latest_release_tag = get_bicep_latest_release_tag()
+            latest_version = _extract_semver(latest_release_tag)
+            if installed_version and latest_version and semver.compare(installed_version, latest_version) < 0:
+                _logger.warning(
+                    'A new Bicep release is available: %s. Upgrade now by running "az bicep upgrade".',
+                    latest_release_tag,
+                )
+
+    return _run_command(installation_path, args)
+
+
+def ensure_bicep_installation(release_tag=None):
+    system = platform.system()
+    installation_path = _get_bicep_installation_path(system)
+
+    if os.path.isfile(installation_path):
+        if not release_tag:
+            return
+
+        installed_version = _get_bicep_installed_version(installation_path)
+        target_version = _extract_semver(release_tag)
+        if installed_version and target_version and semver.compare(installed_version, target_version) == 0:
+            return
+
+    installation_dir = os.path.dirname(installation_path)
+    if not os.path.exists(installation_dir):
+        os.makedirs(installation_dir)
+
+    try:
+        release_tag = release_tag if release_tag else get_bicep_latest_release_tag()
+        if release_tag:
+            print(f"Installing Bicep CLI {release_tag}...")
+        else:
+            print("Installing Bicep CLI...")
+
+        request = urlopen(_get_bicep_download_url(system, release_tag))
+        with open(installation_path, "wb") as f:
+            f.write(request.read())
+
+        os.chmod(installation_path, os.stat(installation_path).st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
+
+        print(f'Successfully installed Bicep CLI to "{installation_path}".')
+    except IOError as err:
+        raise CLIError(f"Error while attempting to download Bicep CLI: {err}")
```

Comment:
Could you change those `CLIError` to some more specific error?
Please refer to `azure.cli.core.azclierror`

Reply:
Changed them to more specific error types.